### PR TITLE
[TT-13142] Fix panic when detailed analytics is turned on with SSE streaming

### PR DESCRIPTION
**Pull request by jeffy-mathew**
## Description
This PR introduces a fix for panic when detailed recording enabled while SSE streaming response.
* Updates opentelemetry library with latest master
* Add nil check for response body in RecordHit

## Related Issue
https://tyktech.atlassian.net/browse/TT-13142

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

___

### **PR Type**
Bug fix

___

### **Description**
- Added a nil check for `responseCopy.Body` in `gateway/handler_success.go` to prevent potential nil pointer dereference errors.
- This change ensures that operations on the response body are only performed if the body is not nil, enhancing the stability of the code.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>handler_success.go</strong><dd><code>Add nil check for responseCopy.Body to prevent errors</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

gateway/handler_success.go

<li>Added a nil check for <code>responseCopy.Body</code> to prevent potential nil <br>pointer dereference.<br> <li> Ensures that the body of the response is not nil before proceeding <br>with operations.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk/pull/6727/files#diff-45135957493eca37f2e3e9a81079577777133c53b27cf95ea2ff0329c05bd006">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

**Comment by buger** [comment]
<p> A JIRA Issue ID is missing from your branch name, PR title and PR description! 🦄</p>
<p>Your branch: fix/sse-streaming</p>
<p>Your PR title: do nil check for responseCopy.Body</p>
<p>Your PR description: 

## Description



## Related Issue






## Motivation and Context



## How This Has Been Tested






## Screenshots (if appropriate)

## Types of changes



- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist





- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why
</p>
<p>If this is your first time contributing to this repository - welcome!</p>
<hr />
<p>Please refer to <a href="https://github.com/cleartax/jira-lint">jira-lint</a> to get started.
<p>Without the JIRA Issue ID in your branch name you would lose out on automatic updates to JIRA via SCM; some GitHub status checks might fail.</p>
Valid sample branch names:

  ‣ feature/shiny-new-feature--mojo-10'
  ‣ 'chore/changelogUpdate_mojo-123'
  ‣ 'bugfix/fix-some-strange-bug_GAL-2345'

**Review by titpetric — APPROVED** [review]

**Review by mativm02 — APPROVED** [review]
LGTM

File: internal/graphql/otel_graphql_engine_detailed_test.go [hunk 1]
```diff
@@ -443,6 +443,10 @@ type spanWrapper struct {
 	attributes map[string][]attribute.KeyValue
 }
 
+func (s *spanWrapper) AddLink(_ trace.Link) {
+	//TODO implement me
+}
+
```
Comment (mativm02) [inline internal/graphql/otel_graphql_engine_detailed_test.go:449]:
```suggestion
func (s *spanWrapper) AddLink(link trace.Link) {
	s.span.AddLink(link)
}
```

File: gateway/testutil.go [hunk 2]
```diff
@@ -2031,3 +2033,69 @@ func (m *testMessageAdapter) Channel() (string, error) {
 func (m *testMessageAdapter) Payload() (string, error) {
 	return m.Msg, nil
 }
+
+func TestHelperSSEServer(tb testing.TB) *httptest.Server {
+	tb.Helper()
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```
Comment (titpetric) [inline gateway/testutil.go:2039]:
```suggestion
	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
```

Comment (titpetric) [reply gateway/testutil.go:2039]: why doesn't golangcilint catch this in pre-commit?

Comment (jeffy-mathew) [reply gateway/testutil.go:2039]: need to check lint rules in that case.

**Pull request merged**